CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 cmake_minimum_required (VERSION 2.8.12)
 project (PIO C Fortran)
 
+# The project version number.
+set(VERSION_MAJOR   2   CACHE STRING "Project major version number.")
+set(VERSION_MINOR   3   CACHE STRING "Project minor version number.")
+set(VERSION_PATCH   8   CACHE STRING "Project patch version number.")
+mark_as_advanced(VERSION_MAJOR VERSION_MINOR VERSION_PATCH)
+
```

Comment:
Strongly suggest you use . instead of _ in the version number. It will make
distributions much easier.

The best end-state is that the ParallelIO library is made available on
standard Linux distributions. This is already the case for netCDF-4 and
HDF5, compiled with mpicc and built with MPICH. So it's possible to set up
the full parallel development environment for netCDF-4 using only yum. Very
handy for users and developers! Yum-based installs prevent support emails
asking about the build (which *is* a tricky build, with lots of mistake
that can be made along the way.)

Of course, on HPC there will always be a lot of building from source, but
programmer's workstations are usually Linux.

The closer we adhere to the standard Linux distribution conventions, the
easier it is to get others to help distribute the library.

Thanks,
Ed

On Wed, Dec 2, 2015 at 9:55 AM, jedwards4b <notifications@github.com> wrote:

> In CMakeLists.txt
> <https://github.com/NCAR/ParallelIO/pull/37#discussion_r46423094>:
>
> > @@ -1,6 +1,12 @@
> >  cmake_minimum_required (VERSION 2.8.12)
> >  project (PIO C Fortran)
> >
> > +# The project version number.
> > +set(VERSION_MAJOR   2   CACHE STRING "Project major version number.")
> > +set(VERSION_MINOR   3   CACHE STRING "Project minor version number.")
> > +set(VERSION_PATCH   8   CACHE STRING "Project patch version number.")
> > +mark_as_advanced(VERSION_MAJOR VERSION_MINOR VERSION_PATCH)
> > +
>
> The current version (tag) in git is 2_0_26 (there is some legacy reason
> that i've long forgotten for using _ instead of . we can probably change
> that) Is there a way to automatically have that git tag inserted here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NCAR/ParallelIO/pull/37/files#r46423094>.
>